command.go

```diff
@@ -479,41 +485,46 @@ func (c *Command) exec(ctx context.Context, opts *CreateOptions, idx int) error
 
 	addOutOp := func(msg message.Fields) message.Fields { return msg }
 	var err error
-	var newProc Process
 	// TODO: the logic below this is not strictly correct if, for example,
 	// Output is redirected to Error and Error has been defined.
 	if opts.Output.Output == nil {
 		var out bytes.Buffer
 		opts.Output.Output = &out
 		opts.Output.Error = &out
-		newProc, err = c.makep(ctx, opts)
+		c.proc, err = c.makep(ctx, opts)
 		if err != nil {
 			return errors.Wrapf(err, "problem starting command")
 		}
 
-		c.procIDs = append(c.procIDs, newProc.ID())
+		c.procIDs = append(c.procIDs, c.proc.ID())
 		addOutOp = func(msg message.Fields) message.Fields {
 			getLogOutput(out.Bytes())
 			return msg
 		}
 	} else {
-		newProc, err = c.makep(ctx, opts)
+		c.proc, err = c.makep(ctx, opts)
 		if err != nil {
 			return errors.Wrapf(err, "problem starting command")
 		}
 
-		c.procIDs = append(c.procIDs, newProc.ID())
+		c.procIDs = append(c.procIDs, c.proc.ID())
 	}
 
 	if !c.runBackground {
-		_, err = newProc.Wait(ctx)
+		_, err = c.proc.Wait(ctx)
 		msg["err"] = err
 	}
-
 	grip.Log(c.priority, addOutOp(msg))
+
 	return errors.WithStack(err)
 }
 
+// Wait returns the exit code and error waiting for the underlying process to
+// complete. This is undefined for Commands run with RunParallel.
+func (c *Command) Wait(ctx context.Context) (int, error) {
+	return c.proc.Wait(ctx)
```

Comment:
I'd have this call some function that returns the process and call wait on that

Reply:
I changed this to keep a slice of procs instead of just one to handle `Run` and `RunParallel` identically. The `Run` case should be the same as before (just wait on the one proc), and the `RunParallel` case should handle waiting on all the procs now.